src/session/session_api.c

```diff
@@ -1682,6 +1682,26 @@ __session_query_timestamp(WT_SESSION *wt_session, char *hex_timestamp, const cha
     API_END_RET(session, ret);
 }
 
+/*
+ * __session_reset_snapshot --
+ *     WT_SESSION->reset_snapshot method.
+ */
+static int
+__session_reset_snapshot(WT_SESSION *wt_session)
+{
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    /* Return error if the isolation mode is read committed. */
+    if (session->txn->isolation != WT_ISO_SNAPSHOT)
+        WT_RET_MSG(session, EPERM, "read committed/uncommitted isolation cannot reset snapshot.");
+
+    __wt_txn_release_snapshot(session);
```

Comment:
The reset snapshot can be executed only the read-only sessions/transactions. Raise an error if the session has performed any write operation.

Reply:
This seems important, ie to return an error if a write operation has been performed. Also, we should include that as a test in the python test.